src/Spark.Engine/Service/FhirServiceExtensions/PatchApplicationService.cs

```diff
@@ -0,0 +1,220 @@
+﻿namespace Spark.Engine.Service.FhirServiceExtensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using Hl7.Fhir.Language;
+    using Hl7.Fhir.Model;
+    using Hl7.FhirPath;
+    using Expression = System.Linq.Expressions.Expression;
+    using fhirExpression = Hl7.FhirPath.Expressions;
+
+    public class PatchApplicationService : IPatchApplicationService
+    {
+        private readonly FhirPathCompiler _compiler;
+
+        public PatchApplicationService()
+        {
+            _compiler = new FhirPathCompiler();
+        }
+
+        public Resource Apply(Resource resource, Parameters patch)
+        {
+            foreach (var component in patch.Parameter.Where(x => x.Name == "operation"))
+            {
+                var operationType = component.Part.First(x => x.Name == "type").Value.ToString();
+                var path = component.Part.First(x => x.Name == "path").Value.ToString();
+                var name = component.Part.FirstOrDefault(x => x.Name == "name")?.Value.ToString();
+                var value = component.Part.FirstOrDefault(x => x.Name == "value")?.Value ?? component.Part.FirstOrDefault(x => x.Name == "value")?.Part[0].Value;
+                var valueExpression = System.Linq.Expressions.Expression.Constant(value);
```

Comment:
Can we refactor this part. Would like to work on a data structure instead of multiple separate variables.

Reply:
Not sure I can. The individual parts are identified by their name. Since I cannot assume an order of the parts based on the spec, I have to find them by their name.

Do you have a suggestion how improve?

Reply:
My intention was to have it refactored somehow so it would be more easily readable, but I haven't read the Patch-specification deeply enough at the moment to have any good suggestions currently

Let's leave it for now